File: pinyougou-cart-interface/src/main/java/com/pinyougou/service/CartService.java
package com.pinyougou.service;

import com.pinyougou.compositeEntity.Cart;

import java.util.List;

/**
 * @author:Jack
 * @description:
 * @Date:created in 2018/10/25,13:27
 */
public interface CartService {

    public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);

    public List<Cart> selectCartListFromRedis(String username);

    public void saveCartListToRedis(String username,List<Cart> cartList);

    public  List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService', function ($http) {
    //购物车列表
    this.selectCartListFromCookie = function () {
        return $http.get('cart/selectCartListFromCookie.do');
    };
    //更改购物车商品数
    this.addGoodsToCartList = function (itemId, num) {
        return $http.get('cart/addGoodsToCartList.do?itemId=' + itemId + '&num=' + num);
    };
    //获取购物车总商品数及总价
    this.sum = function (cartList) {
        var total = {"totalNum": 0, "totalMoney": 0};
        for (var i = 0; i < cartList.length; i++) {
            var orderItemList = cartList[i].orderItemList;
            for (var j = 0; j < orderItemList.length; j++) {
                total.totalNum += orderItemList[j].num;
                total.totalMoney += orderItemList[j].totalFee;
            }
        }
    return total;
    }


});